src/org/javarosa/core/model/instance/TreeReference.java

```diff
@@ -499,13 +503,14 @@ public String toString (boolean includePredicates) {
 
             if (includePredicates) {
                 switch (mult) {
-                case INDEX_UNBOUND: break;
-                case INDEX_TEMPLATE: sb.append("[@template]"); break;
-                case INDEX_REPEAT_JUNCTURE: sb.append("[@juncture]"); break;
-                default:
-                    if ((i > 0 || mult != 0) && mult !=-4)
-                        sb.append("[" + (mult + 1) + "]");
-                    break;
+                    case INDEX_UNBOUND: break;
+                    case INDEX_TEMPLATE: sb.append("[@template]"); break;
+                    case INDEX_REPEAT_JUNCTURE: sb.append("[@juncture]"); break;
+                    default:
+                        if ((i > 0 || mult != 0) && mult !=-4) {
+                            sb.append("[").append(zeroIndexMult ? mult : mult + 1).append("]");
```

Comment:
It’s fine how it is, but in case you didn’t think of it and prefer:

append(mult + (zeroIndexMult ? 0 : 1))